Library/Homebrew/cmd/help.rb

```diff
@@ -39,4 +39,24 @@ def help
   def help_s
     HOMEBREW_HELP
   end
+
+  def help_cmd(cmd)
+    if File.exist?(HOMEBREW_LIBRARY_PATH/"cmd/#{cmd}.sh")
+      cmd_path = HOMEBREW_LIBRARY_PATH/"cmd/#{cmd}.sh"
+    elsif ARGV.homebrew_developer? && File.exist?(HOMEBREW_LIBRARY_PATH/"dev-cmd/#{cmd}.sh")
+      cmd_path = HOMEBREW_LIBRARY_PATH/"dev-cmd/#{cmd}.sh"
+    elsif File.exist?(HOMEBREW_LIBRARY_PATH/"cmd/#{cmd}.rb")
+      cmd_path = HOMEBREW_LIBRARY_PATH/"cmd/#{cmd}.rb"
+    elsif ARGV.homebrew_developer? && File.exist?(HOMEBREW_LIBRARY_PATH/"dev-cmd/#{cmd}.rb")
+      cmd_path = HOMEBREW_LIBRARY_PATH/"dev-cmd/#{cmd}.rb"
+    else
+      return ""
+    end
```

Comment:
I reckon you may be able to combine it with some of the existing internal command logic, too.